src/components/view_article_page/CommentPoster.jsx

```diff
@@ -0,0 +1,48 @@
+import { useState, useContext } from "react"
+import { UserContext } from "../../contexts/UserContext"
+import { ErrorContext } from "../../contexts/ErrorContext"
+import { postComment } from "../../api"
+
+export default function CommentPoster({article_id, setCommentList}) {
+    const [comment, setComment] = useState("")
+    const { user } = useContext(UserContext)
+    const {error, setError} = useContext(ErrorContext)
+
+    function handleChange(event) {
+        setComment(event.target.value)
+    }
+
+    function handleSubmit(event) {
+        event.preventDefault()
+
+        const body = comment.trim()
+
+        postComment(article_id, user.username, body)
+        .then((newComment) => {
+            setCommentList((commentList) => {
+                return [newComment, ...commentList]
+            })
+            setComment("")
+        })
+        .catch(() => {
+            setError(currError => {
+                const newError = {...currError}
+                newError.msg = "Failed to post comment try again later"
+                newError.show = true
+                return newError
```

Comment:
Nice work handling the error, but at the moment when I do an offline post, I don't seem to get the error message showing up? Could we have something in our html that displays a message for the user if it's been unsuccessful please?